docs/tutorials/getting-started-with-flask.rst

```diff
@@ -0,0 +1,114 @@
+Getting started with Rockcraft for Flask
+----------------------------------------
+
+In this tutorial you will learn how to create a Flask application, build a rock
+with the Flask application and run it.
+
+Prerequisites
+-------------
+- snap enabled system (https://snapcraft.io)
+- LXD installed (https://linuxcontainers.org/lxd/getting-started-cli/)
+- Docker installed (https://snapcraft.io/docker)
+- a text editor
+
+Create Flask application
+------------------------
+
+Create a new directory and write the following into a text editor and save it as
+``requirements.txt``:
+
+.. literalinclude:: code/getting-started-with-flask/requirements.txt
+
+Install ``python3-venv`` on your host and create a virtual environment:
+
+.. literalinclude:: code/getting-started-with-flask/task.yaml
+    :language: bash
+    :start-after: [docs:create-venv]
+    :end-before: [docs:create-venv-end]
+    :dedent: 2
+
+In the same directory, write the following into a text editor and save it as
+``app.py``:
+
+.. literalinclude:: code/getting-started-with-flask/app.py
+    :language: python
+
+Run and test the flask application to verify everything is working:
+
+.. literalinclude:: code/getting-started-with-flask/task.yaml
+    :language: bash
+    :start-after: [docs:run-flask]
+    :end-before: [docs:run-flask-end]
+    :dedent: 2
+
+The ``curl`` command should print ``Hello, world!``.
+
+Install Rockcraft
+-----------------
+
+Install Rockcraft on your host:
+
+.. literalinclude:: code/hello-world/task.yaml
+    :language: bash
+    :start-after: [docs:install-rockcraft]
+    :end-before: [docs:install-rockcraft-end]
+    :dedent: 2
+
+Create Flask rock
+-----------------
+
+Use the ``flask-framework`` profile to create ``rockcraft.yaml``:
+
+.. literalinclude:: code/getting-started-with-flask/task.yaml
+    :language: bash
+    :start-after: [docs:create-rockcraft-yaml]
+    :end-before: [docs:create-rockcraft-yaml-end]
+    :dedent: 2
+
+Open ``rockcraft.yaml`` in a text editor and customise the ``name``,
```

Comment:
I think these steps should be dumbed down a bit more. e.g. the `sed` above isn't as important as the `rockcraft init`. and it's only being done for the sake of this tutorial.

Reply:
removed